datahub-frontend/app/controllers/AuthenticationController.java

```diff
@@ -292,7 +292,8 @@ private Optional<Result> redirectToIdentityProvider(Http.RequestHeader request,
 
     private void configurePac4jSessionStore(PlayWebContext context, Client client, String redirectPath) {
         // Set the originally requested path for post-auth redirection.
-        _playSessionStore.set(context, Pac4jConstants.REQUESTED_URL, new FoundAction(redirectPath));
+        // Commenting this to avoid https://github.com/datahub-project/datahub/issues/7221
+        // _playSessionStore.set(context, Pac4jConstants.REQUESTED_URL, new FoundAction(redirectPath));
```

Comment:
Looks like this is failing linting

Reply:
yes. With the commented lines, there were unused imports. fixed

Reply:
It's not clear to me .. why this would be causing a 502.. Is the direct path just ... incorrect?

Reply:
Or is it that the path is just too large? Or something else? Or could it be that the path is simply invalid? Does it appear that the browser is getting bad a 302?

Would it be possible that your reverse proxy (nginx) is not able to propagate a redirect (302) up to the browser correctly?